sync/interface.go.go

```diff
@@ -0,0 +1,24 @@
+package sync
+
+import (
+	"context"
+	"io"
+)
+
+// Interface is used by Testground components to enable mocking/stubbing in
+// tests. Consequently, it doesn't have the Must* variants, as those are useful
+// for test plans only.
+//
+// If you're a test plan writer, you should not be using this.
+type Interface interface {
```

Comment:
`Interface` for a name sounds weird. Wouldn't it be better to call this `Service` ? Basically then we have multiple impl. of the `sync.Service` ?

Reply:
But it's not really the _service_... I see why the name can be ugly, but it's idiomatic in go: https://golang.org/pkg/sort/#Interface :-/

Reply:
It is the interface of the sync service - precisely what we a mocking for in-memory usage, no? In any case I don't have strong feelings on how we call this.

Reply:
Yeah, but introducing the term `Service` to abstract over a `Client` is rather weird. Those names tend to have well-defined expectations.

Reply:
I agree, that this is really the `Client` of the `sync service`. At the moment we don't have a separate process for the `sync service` concept, it is really just a library interacting with Redis... This is why I think of the `Client` and the `Service` as the same thing right now.

In the future, if we split the `sync service` into a separate component, and it becomes not just a library that testplans import, but rather import a `Client` that interacts with the `Service` - then they would really be different things.

Reply:
Yup, in that case it's very likely that the `Service` will not live in the SDK, even.

Reply:
I could see this being called `Syncer` or something. Most of the time when I see an interface simply called Interface, it is an interface to be implemented. i.e. sync.Interface, heap.Interface. Either way works for me though @nonsense if you have an opinion about it we can change this.